core/src/test/scala/sttp/openai/requests/files/FilesResponseDataSpec.scala

```diff
@@ -34,4 +34,26 @@ class FilesResponseDataSpec extends AnyFlatSpec with Matchers with EitherValues
     // then
     givenResponse.value shouldBe expectedResponse
   }
+
+  "Given upload file response as Json" should "be properly deserialized to case class" in {
+    // given
+    val singleFileJsonResponse = fixtures.FilesResponse.singleFileJsonResponse
+    val expectedResponse =
+      FileData(
+        `object` = "file",
+        id = "file-tralala",
+        purpose = "fine-tune",
+        filename = "example.jsonl",
+        bytes = 44,
+        createdAt = 1681375533,
+        status = "uploaded",
+        statusDetails = None
+      )
+
+    // when
+    val givenResponse: Either[Exception, FileData] = SttpUpickleApiExtension.deserializeJsonSnake[FileData].apply(singleFileJsonResponse)
```

Comment:
**question:** Is `Exception` the strongest error we can get here?

Reply:
Ok, I checked upickle sources and the API indeed can throw a few kinds of `Exception`, some of them upickle-specific, sometimes raw Java exceptions like `IllegalArgumentException`, which is not nice.
However, I noticed in SnakePickle, that you're handling errors there with:
```scala
    try
      Right(SnakePickle.read[B](JsonInput.sanitize[B].apply(s)))
    catch {
      case e: Exception => Left(e)
      case t: Throwable =>
        // in ScalaJS, ArrayIndexOutOfBoundsException exceptions are wrapped in org.scalajs.linker.runtime.UndefinedBehaviorError
        t.getCause match {
          case e: ArrayIndexOutOfBoundsException => Left(e)
          case _                                 => throw t
        }
    }
```
How about changing both occurrences of `Left(e)` to `Left(new DeserializationException(e)`, where `DesrializationException` will be your own type, defined in `sttp.openai.json`? This could allow slightly better assertions here, wdyt?